src/Parsers/ASTAlterQuery.cpp

```diff
@@ -127,364 +127,368 @@ const char * ASTAlterCommand::typeToString(ASTAlterCommand::Type type)
     UNREACHABLE();
 }
 
-void ASTAlterCommand::formatImpl(const FormatSettings & settings, FormatState & state, FormatStateStacked frame) const
+void ASTAlterCommand::formatImpl(FormattingBuffer out) const
 {
     if (type == ASTAlterCommand::ADD_COLUMN)
     {
-        settings.ostr << (settings.hilite ? hilite_keyword : "") << "ADD COLUMN " << (if_not_exists ? "IF NOT EXISTS " : "")
-                      << (settings.hilite ? hilite_none : "");
-        col_decl->formatImpl(settings, state, frame);
+        out.writeKeyword("ADD COLUMN ");
+        out.writeKeyword(if_not_exists ? "IF NOT EXISTS " : "");
+        col_decl->formatImpl(out);
 
         if (first)
-            settings.ostr << (settings.hilite ? hilite_keyword : "") << " FIRST " << (settings.hilite ? hilite_none : "");
+            out.writeKeyword(" FIRST ");
         else if (column) /// AFTER
         {
-            settings.ostr << (settings.hilite ? hilite_keyword : "") << " AFTER " << (settings.hilite ? hilite_none : "");
-            column->formatImpl(settings, state, frame);
+            out.writeKeyword(" AFTER ");
+            column->formatImpl(out);
         }
     }
     else if (type == ASTAlterCommand::DROP_COLUMN)
     {
-        settings.ostr << (settings.hilite ? hilite_keyword : "") << (clear_column ? "CLEAR " : "DROP ") << "COLUMN "
-                      << (if_exists ? "IF EXISTS " : "") << (settings.hilite ? hilite_none : "");
-        column->formatImpl(settings, state, frame);
+        out.writeKeyword(clear_column ? "CLEAR " : "DROP ");
+        out.writeKeyword("COLUMN ");
+        out.writeKeyword(if_exists ? "IF EXISTS " : "");
```

Comment:
We don't have to call writeKeyword() if if_not_exists == false

Reply:
Same, won't fix